standalone/standalone.tpl

```diff
@@ -1951,112 +2003,145 @@ def train(
+                                    if (
+                                        container_status.state.waiting
+                                        and container_status.state.waiting.reason
+                                        == "CrashLoopBackOff"
+                                    ):
+                                        log_pod_containers(
+                                            pod_event, "init_containers", namespace
+                                        )
+                                        log_pod_containers(
+                                            pod_event, "containers", namespace
+                                        )
+                                        raise RuntimeError(
+                                            f"Pod {pod_event.metadata.name} failed."
+                                        )
+                                if pod_event.status.phase == "Failed":
+                                    log_pod_containers(
+                                        pod_event, "init_containers", namespace
+                                    )
+                                    log_pod_containers(
+                                        pod_event, "containers", namespace
+                                    )
+                                    w.stop()
+                                if pod_event.status.phase == "Succeeded":
+                                    if pod_event.metadata.name.startswith(
+                                        f"{pytorchjob_name}-master"
+                                    ):
+                                        master_pod_success = True
+                                        logger.info(
+                                            "Pod '%s' completed successfully",
+                                            pod_event.metadata.name,
+                                        )
+                                    elif pod_event.metadata.name.startswith(
+                                        f"{pytorchjob_name}-worker"
+                                    ):
+                                        logger.info(
+                                            "Pod '%s' completed successfully",
+                                            pod_event.metadata.name,
+                                        )
+                                        worker_pod_success = True
+                                    if master_pod_success and worker_pod_success:
+                                        logger.info(
+                                            "All PytorchJob Pods completed successfully"
+                                        )
+                                        w.stop()
+                                        exit_flag = True
+                                        # Break here to avoid going into other conditions, we are
+                                        # done
+                                        break
```

Comment:
I won't block merging on this as we want to get this watcher issue resolved, but it's worth noting here: There will always be a single master pod, but in an actual situation there might be many more worker pods. This looks like it will trigger `w.stop()` and `exit_flag = True` once the first of N worker pods completes. But we actually want to wait for all worker pods to finish first.

Reply:
Good point, will address next.

Reply:
https://github.com/redhat-et/ilab-on-ocp/pull/103